core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -104,13 +104,23 @@ sealed abstract class Aggregate extends Data {
 
   private[chisel3] def width: Width = getElements.map(_.width).foldLeft(0.W)(_ + _)
 
-  private[chisel3] def legacyConnect(that: Data)(implicit sourceInfo: SourceInfo): Unit = {
+  private[chisel3] def firrtlConnect(that: Data)(implicit sourceInfo: SourceInfo): Unit = {
     // If the source is a DontCare, generate a DefInvalid for the sink,
     //  otherwise, issue a Connect.
     if (that == DontCare) {
-      pushCommand(DefInvalid(sourceInfo, Node(this)))
+      pushCommand(DefInvalid(sourceInfo, lref))
     } else {
-      pushCommand(BulkConnect(sourceInfo, Node(this), Node(that)))
+      pushCommand(Connect(sourceInfo, lref, Node(that)))
+    }
+  }
+
+  private[chisel3] def firrtlPartialConnect(that: Data)(implicit sourceInfo: SourceInfo): Unit = {
```

Comment:
note that this was the thing previously called "legacyConnect". I decided to just describe what it is doing and call it `firrtlPartialConnect` instead of the historical path that we got here. Thoughts?

I am happy to roll back the `BulkConnect -> PartialConnect` change though I think its extremely confusing to have BulkConnect basically mean opposite things in Chisel and Firrtl